### PR TITLE
Use deployed contracts in fixtures

### PR DESCRIPTION
**Pull request by PeterYinusa**
## Explanation

See: #17752

This PR allows the smart contract deployed in the test to be used inside the test fixture.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: test/e2e/fixture-builder.js [hunk 1]
```diff
@@ -1321,15 +1397,6 @@ class FixtureBuilder {
     });
   }
 
-  withNetworkSupportEIP1559() {
```
Comment (PeterYinusa) [inline test/e2e/fixture-builder.js:1324]: Renamed `withNetworkSupportEIP1559` to `withNetworkControllerSupportEIP1559` so its easy to find, and sorted the file alphabetically

File: test/e2e/fixture-builder.js [hunk 2]
```diff
@@ -495,6 +486,91 @@ class FixtureBuilder {
...
+    merge(
+      this.fixture.data.NftController
+        ? this.fixture.data.NftController
+        : (this.fixture.data.NftController = {}),
+      data,
+    );
+    return this;
+  }
+
+  withNftControllerERC1155() {
```
Comment (PeterYinusa) [inline test/e2e/fixture-builder.js:508]: Add deployed ERC1155 fixture data

File: test/e2e/fixture-builder.js [hunk 3]
```diff
@@ -495,6 +486,91 @@ class FixtureBuilder {
...
+              standard: 'ERC1155',
+            },
+          ],
+        },
+      },
+      ignoredNfts: [],
+    });
+  }
+
+  withNftControllerERC721() {
```
Comment (PeterYinusa) [inline test/e2e/fixture-builder.js:540]: Add deployed ERC721 fixture data

File: test/e2e/helpers.js [hunk 4]
```diff
@@ -75,7 +75,7 @@ async function withFixtures(options, testSuite) {
       });
     }
     await fixtureServer.start();
-    fixtureServer.loadJsonState(fixtures);
+    fixtureServer.loadJsonState(fixtures, contractRegistry);
```
Comment (PeterYinusa) [inline test/e2e/helpers.js:78]: Share the smart contract registry with the fixture server

File: test/e2e/fixture-server.js [hunk 5]
```diff
@@ -35,6 +41,9 @@ function performSubstitution(partialState) {
       fixtureSubstitutionCommands.currentDateInMilliseconds
     ) {
       return new Date().getTime();
+    } else if (partialState.includes(CONTRACT_KEY)) {
+      const contract = partialState.split(CONTRACT_KEY).pop();
+      return contractRegistry.getContractAddress(contract);
     }
```
Comment (PeterYinusa) [inline test/e2e/fixture-server.js:47]: Replace fixture `__FIXTURE_SUBSTITUTION__CONTRACT${SMART_CONTRACTS.NFTS}` with `0x581c3C1A2A4EBDE2A0Df29B5cf4c116E42945947`

File: test/e2e/nft/erc1155-interaction.spec.js [hunk 6]
```diff
@@ -101,6 +105,11 @@ describe('ERC1155 NFTs testdapp interaction', function () {
           windowHandles,
         );
 
+        // Confirm Transfer
+        await driver.waitForSelector({
+          css: '.confirm-page-container-summary__action__name',
+          text: 'Deposit',
+        });
         await driver.clickElement({ text: 'Confirm', tag: 'button' });
```
Comment (PeterYinusa) [inline test/e2e/nft/erc1155-interaction.spec.js:113]: same as above

File: test/e2e/nft/remove-nft.spec.js [hunk 7]
```diff
@@ -19,27 +19,18 @@ describe('Remove NFT', function () {
...
 
-        // After login, go to NFTs tab and import an NFT
-        await driver.clickElement('[data-testid="home__nfts-tab"]');
-        await driver.clickElement({ text: 'Import NFTs', tag: 'a' });
-
-        await driver.fill('[data-testid="address"]', contractAddress);
-        await driver.fill('[data-testid="token-id"]', '1');
-        await driver.clickElement({ text: 'Add', tag: 'button' });
-
         // Open the details and click remove nft button
```
Comment (PeterYinusa) [inline test/e2e/nft/remove-nft.spec.js:32]: We now import the ERC721 using the fixture. We have separate tests for the import flow

File: test/e2e/nft/send-nft.spec.js [hunk 8]
```diff
@@ -19,27 +19,18 @@ describe('Send NFT', function () {
...
         await driver.press('#password', driver.Key.ENTER);
 
-        // After login, go to NFTs tab and import an NFT
-        await driver.clickElement('[data-testid="home__nfts-tab"]');
-        await driver.clickElement({ text: 'Import NFTs', tag: 'a' });
-
-        await driver.fill('[data-testid="address"]', contractAddress);
-        await driver.fill('[data-testid="token-id"]', '1');
-        await driver.clickElement({ text: 'Add', tag: 'button' });
-
```
Comment (PeterYinusa) [inline test/e2e/nft/send-nft.spec.js:41]: same as above

File: test/e2e/nft/view-nft-details.spec.js [hunk 9]
```diff
@@ -21,27 +21,18 @@ describe('View NFT details', function () {
...
 
-        // After login, go to NFTs tab and import an NFT
-        await driver.clickElement('[data-testid="home__nfts-tab"]');
-        await driver.clickElement({ text: 'Import NFTs', tag: 'a' });
-
-        await driver.fill('[data-testid="address"]', contractAddress);
-        await driver.fill('[data-testid="token-id"]', '1');
-        await driver.clickElement({ text: 'Add', tag: 'button' });
-
         // Click to open the NFT details page and check title
```
Comment (PeterYinusa) [inline test/e2e/nft/view-nft-details.spec.js:34]: same as above

File: test/e2e/tests/send-eth.spec.js [hunk 10]
```diff
@@ -302,7 +302,7 @@ describe('Send ETH from dapp using advanced gas controls', function () {
         dapp: true,
         fixtures: new FixtureBuilder()
           .withPermissionControllerConnectedToTestDapp()
-          .withNetworkSupportEIP1559()
```
Comment (PeterYinusa) [inline test/e2e/tests/send-eth.spec.js:305]: This was renamed to follow the naming convention in the file

**Review by chloeYue — APPROVED** [review]
Great improvement, LGTM, thanks!

**Review by seaona — APPROVED** [review]
Looks great! I appreciate the review comments for following alongside the changes :heart:

**Pull request merged**